scraper-tools/src/browser_tools.js

```diff
@@ -165,17 +165,19 @@ const saveSnapshot = async ({ page, $ }) => {
     }
     lastSnapshotTimestamp = now;
 
-    if ($) {
-        await Apify.setValue(SNAPSHOT.KEYS.HTML, $.html(), { contentType: 'text/html' });
-    }
-    if (page) {
+    if (body && contentType) {
+        const rawBody = (typeof body === 'object') ? JSON.stringify(body) : body;
```

Comment:
Is this correct? What if body is an image (buffer) then `typeof body === 'object'` holds and it gets to a loop, this should be done based on content type